pkg/rke2/rke2.go

```diff
@@ -36,6 +38,12 @@ func Server(ctx *cli.Context, cfg Config) error {
 	if err := ctx.Set("secrets-encryption", "true"); err != nil {
 		return err
 	}
+	go func() {
```

Comment:
Im not sure about this being in a goroutine. I feel like maybe it should block, but I'll seek a second opinion

Reply:
I remember we discussed that but that function call is in k3s that is responsible for a few setup routines that the new code is dependent on. The new code has checks and waits for it to be put in place. eg `/etc/rancher/rke2/rke2.yaml`.

Reply:
Seems like the right approach to me, re: the goroutine.

Reply:
must-have - but needs a conversation with Darren.

This is really a bigger conversation about how RKE2 can tie into the server bootstrapping process in a clean way. This sets the precedence for how we do that. Right now, the k3s code exposes no hooks for doing this cleanly.

Reply:
Updated.